src/components/WeatherForm.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import { fetchWeather } from '../utils'
+import WeatherResults from './WeatherResults'
+
+class WeatherForm extends Component{
+  state = {
+    userRegionInput: '',
+    regionName: '',
+    country: '',
+    weather: '',
+    description: '',
+    temp_max: '',
+    temp_min: '',
+    weatherIcon: '',
+    error: ''
+  }
+
+
+  handleChange(event){
+    const {name, value} = event.target
+
+    this.setState({
+      [name]: value
+    });
+  }
+
+  async handleSubmit(evt){
+    evt.preventDefault()
+
+    const userRegionInput = this.state.userRegionInput
+
+    let response = await fetchWeather(userRegionInput)
+
+    if(response){
+
+      let regionName = response.name;
+      let country = response.sys.country;
+      let weather = response.weather[0].main;
+      let description = response.weather[0].description;
+      let temp_max = response.main.temp_max;
+      let temp_min = response.main.temp_min;
+      let weatherIcon = response.weather[0].icon;
+
+      this.setState({
+        userRegionInput: '',
+        description: description,
+        weatherIcon: weatherIcon,
+        regionName: regionName,
+        country: country,
+        weather: weather,
+        temp_max: temp_max,
+        temp_min: temp_min,
+        error: ''
+      })
+      
+    } else {
+      this.setState({
+        error: 'Region Not Found. Please Check Your Input'
+      })
+    }
+  }
+
+  render(){
+    const {
+      userRegionInput,
+      regionName,
+      country,
+      weather,
+      description,
+      temp_max,
+      temp_min,
+      weatherIcon,
+      error } = this.state
+
+    return(
+      <section>
+        <p>Enter your city or state</p>
+
+        <form onSubmit={(evt) => this.handleSubmit(evt)}>
+          <input
+            type='text'
+            name='userRegionInput'
+            value={userRegionInput}
+            onChange={(event) =>
+              this.handleChange(event)}
+            placeholder='Enter Here'
+            autoComplete='off'/>
+        </form>
```

Comment:
[suggestion] What do you think about adding a submit button so that the user can either click the buton or hit enter?

Reply:
Yes, That would be the best solution. Specially from an user experience perspective.